### PR TITLE
Improve README

### PR DESCRIPTION
**Pull request by glyn**
The README doesn't look quite right. I propose to use this PR to discuss and tune it.

/cc @listx

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -105,11 +105,11 @@ If there are multiple destination registries, the above calculation is repeated
 for each destination registry. The promoter also prints warnings about images in
 `M` that cannot be found in the source registry (call it `S`):
```
Comment (glyn) [inline README.md]: Does "it" refer to the images in the source registry?

**Comment by glyn** [comment]
I was surprised by the definition of `D` as all images in the destination _registry_. There are typically lots of images that shouldn't be relevant.

Also, I couldn't find the flag `delete-extra-tags` in the code (and I couldn't build the code to try the online help: https://github.com/kubernetes-sigs/k8s-container-image-promoter/issues/121).

File: README.md (hunk 1 above)
Comment (listx) [reply README.md]: > Does "it" refer to the images in the source registry?

Yes.

At the same time, the logic around using Set difference for calculating what needs to be copied over was dropped in https://github.com/kubernetes-sigs/k8s-container-image-promoter/pull/86/commits/b062e7a33898604917394986036d26c15c476562, so the language in the README using these concepts should be removed.

**Comment by listx** [comment]
The README as it is today needs more work. You caught something that has been in the back burner of my mind for a while now.

> I was surprised by the definition of `D` as all images in the destination _registry_. There are typically lots of images that shouldn't be relevant.

Correct. There are 2 things to note here: (1) technically, gcr.io is a registry, and gcr.io/foo is a repository and we only really care about repositories, not registries; (2) the set-difference logic to calculate what needs to be promoted/deleted was removed, and so we no longer care about *all* images in a registry (or repository). I vote for simply removing the explanation via set theory.

> 
> Also, I couldn't find the flag `delete-extra-tags` in the code (and I couldn't build the code to try the online help: #121).

This flag was removed on purpose in https://github.com/kubernetes-sigs/k8s-container-image-promoter/pull/81

**Comment by glyn** [comment]
@listx Thanks for the feedback! The set-based _implementation_ may have gone, but wouldn't a set-theoretic description of the _what_ the implementation does (rather than _how_ it does it) still be worth keeping? I would hope so...

**Comment by listx** [comment]
> @listx Thanks for the feedback! The set-based _implementation_ may have gone, but wouldn't a set-theoretic description of the _what_ the implementation does (rather than _how_ it does it) still be worth keeping? I would hope so...

Hmm, I suppose so.

Meanwhile can you sign the CLA agreement? That's a blocker for this to get merged.

**Comment by glyn** [comment]
> > @listx Thanks for the feedback! The set-based _implementation_ may have gone, but wouldn't a set-theoretic description of the _what_ the implementation does (rather than _how_ it does it) still be worth keeping? I would hope so...
> 
> Hmm, I suppose so.

Let's work that out together in this PR.

> 
> Meanwhile can you sign the CLA agreement? That's a blocker for this to get merged.

I was told that my name had been added to Pivotal's corporate CLA, so I'll follow up to see what else needs doing.

**Comment by glyn** [comment]
Not sure why the CLA check is failing. When I click "details" and try to sign as an employee, it looks ok:
Screenshot 2019-09-16 at 08 49 03

**Comment by glyn** [comment]
@listx The CLA is sorted now. Please could you merge or let me have further comments.

**Comment by dims** [comment]
/approve

leaving `/lgtm` for @listx

**Comment by listx** [comment]
My Github-foo is weak and I don't understand why this PR is called a "draft" PR and cannot be merged. Help?

**Comment by Katharine** [comment]
This is causing tide issues.

/hold


@listx a draft PR is a type of PR that is natively flagged as a draft by github and cannot be merged. The original creator has to remove the flag via the UI.

**Comment by glyn** [comment]
I made this a draft PR because I was expecting more iteration, but if it's good to go, let's merge it.

/lgtm

**Pull request merged**